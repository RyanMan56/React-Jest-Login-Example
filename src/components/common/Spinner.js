import React from 'react';
import ripple from '../../assets/ripple.svg';

const Spinner = (props) => (
  <img {...props} src={ripple} alt="loading-spinner" />
);

export default Spinner;