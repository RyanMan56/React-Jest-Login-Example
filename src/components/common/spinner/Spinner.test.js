import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Spinner from './Spinner';

// Shallowly tests component using jest and enzyme, ignoring any child components
it('shallowly renders without crashing', () => {
  shallow(<Spinner />);
});

it('matches snapshot', () => {
  const spinner = renderer
    .create(
      <Spinner />
    ).toJSON();
  expect(spinner).toMatchSnapshot();
})