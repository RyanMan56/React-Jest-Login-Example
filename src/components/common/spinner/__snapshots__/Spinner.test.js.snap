// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<img
  alt="loading-spinner"
  src="ripple.svg"
/>
`;
